aea/mail/base.py

```diff
@@ -54,171 +55,120 @@ def __init__(self, uri_raw: str) -> None:
-        self._password = parsed.password
-        self._host = parsed.hostname
-        self._port = parsed.port
+        self._uri_raw = uri_raw
 
     @property
     def scheme(self) -> str:
         """Get the scheme."""
-        return self._scheme
+        parsed = urlparse(self._uri_raw)
+        return parsed.scheme
 
     @property
     def netloc(self) -> str:
         """Get the netloc."""
-        return self._netloc
+        parsed = urlparse(self._uri_raw)
+        return parsed.netloc
 
     @property
     def path(self) -> str:
         """Get the path."""
-        return self._path
+        parsed = urlparse(self._uri_raw)
+        return parsed.path
 
     @property
     def params(self) -> str:
         """Get the params."""
-        return self._params
+        parsed = urlparse(self._uri_raw)
+        return parsed.params
 
     @property
     def query(self) -> str:
         """Get the query."""
-        return self._query
+        parsed = urlparse(self._uri_raw)
+        return parsed.query
 
     @property
     def fragment(self) -> str:
         """Get the fragment."""
-        return self._fragment
+        parsed = urlparse(self._uri_raw)
+        return parsed.fragment
 
     @property
     def username(self) -> Optional[str]:
         """Get the username."""
-        return self._username
+        parsed = urlparse(self._uri_raw)
+        return parsed.username
 
     @property
     def password(self) -> Optional[str]:
         """Get the password."""
-        return self._password
+        parsed = urlparse(self._uri_raw)
+        return parsed.password
 
     @property
     def host(self) -> Optional[str]:
         """Get the host."""
-        return self._host
+        parsed = urlparse(self._uri_raw)
+        return parsed.hostname
 
     @property
     def port(self) -> Optional[int]:
         """Get the port."""
-        return self._port
+        parsed = urlparse(self._uri_raw)
+        return parsed.port
 
     def __str__(self) -> str:
         """Get string representation."""
-        return self.uri_raw
+        return self._uri_raw
 
     def __eq__(self, other: Any) -> bool:
         """Compare with another object."""
-        return (
-            isinstance(other, URI)
-            and self.scheme == other.scheme
-            and self.netloc == other.netloc
-            and self.path == other.path
-            and self.params == other.params
-            and self.query == other.query
-            and self.fragment == other.fragment
-            and self.username == other.username
-            and self.password == other.password
-            and self.host == other.host
-            and self.port == other.port
-        )
+        return isinstance(other, URI) and str(self) == str(other)
 
 
 class EnvelopeContext:
-    """Extra information for the handling of an envelope."""
+    """Contains context information of an envelope."""
+
+    __slots__ = ("_connection_id", "_uri")
```

Comment:
is it possible that uri will contain connection_id details?

Reply:
No; we don't allow URI to "target" anything in the AEA. This would be an obvious attack vector.